docs/components/INTRODUCTION.md

```diff
@@ -0,0 +1,73 @@
+# Component introduction
+
+A component refers to a modular, reusable piece of code that encapsulates specific functionality, styling, and behavior within a user interface (UI). Components are commonly used to build dynamic and interactive web applications, allowing developers to break down complex user interfaces into smaller, manageable parts.
+
+## How to generate a component
+
+```shell
+ng g component <ComponentName> [--project-name <project-name>]
+```
+
+## Structure
+Below you can see the basic files generated for a component (more files can be generated depending on the activated options):
+- `component-name.component.ts` will contain the logic of the component
+- `component-name.spec.ts` will contain component unit tests
+- `component-name.style.scss` will contain the styling of the component
+- `component-name.template.html` will contain the structure of the component
+
+> [!NOTE]
+> By convention, file names must be written in kebab-case
+
+
+## Component class
+Otter components are Angular components with Otter-specific information provided through the `@O3rComponent` decorator.
```

Comment:
We could specify what is the use of this extra info? (Customization, debugging etc.)

Reply:
We introduce a decorator but our users won't really know what is this for.